fp/template-engine.js

```diff
@@ -180,6 +190,38 @@ exports.fillTemplates = function (ctx, image, params, blocks, paramIgnoreMap) {
     console.log("       " + (prefix ? prefix + "." : "") +
                 file.replace(/.*\/([^\/]+)$/, "$1") + ": " + txt);
   }
-};
 
+  function getPermission(file){	
+    var pathToFile = file.replace(/\/\//g, '/');
+    var exist = fs.existsSync(pathToFile);
+    if(!exist) {
+       return;
+    }
+
+    var stats = fs.statSync(file);
+    return '0' + (stats.mode & parseInt('777', 8)).toString(8);
+  }
+
+  function setPermission(file, permission) {
+    try {
+     fs.chmodSync(file, permission);
+    }
+    catch(e) {}
```

Comment:
Shouldn't we stop here, as when the permission cant be set, then the output is not as expected ? (so not reproducible)

Reply:
I shift the chmodSync inside the fillTemplate method